### PR TITLE
fix: Solve the problem of failure to download multi-branch-pipeline artifacts

### PR DESCRIPTION
**Pull request by littlejiancc**

### CODE REVIEW

File: pkg/artifact/artifacts.go [hunk 1]
```diff
@@ -37,8 +37,8 @@ func (q *Client) List(jobName string, buildID int) (artifacts []Artifact, err er
 }
 
 // GetArtifact download artifact using stream
-func (q *Client) GetArtifact(projectName, pipelineName string, buildID int, filename string) (io.ReadCloser, error) {
-	artifactURL := fmt.Sprintf("/job/%s/job/%s/%d/artifact/%s", projectName, pipelineName, buildID, filename)
+func (q *Client) GetArtifact(projectName, pipelineName string, isMultiBranch bool, branchName string, buildID int, filename string) (io.ReadCloser, error) {
```
Comment (LinuxSuRen) [inline pkg/artifact/artifacts.go:40]: Consider the compitale, I prefer to provide a new function with the new parameters. For instance:

`func (q *Client) GetArtifactFromBranchPipeline(projectName, pipelineName string, isMultiBranch bool, branchName string, buildID int, filename string) (io.ReadCloser, error) {`

then, `GetArtifact` could call `GetArtifactFromBranchPipeline`.

The other developers might occur the compile errors if we just add some new parameters.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Thanks for helping on this project. Please feel free to see the comments.

File: pkg/artifact/artifacts.go (hunk 1 above)
Comment (littlejiancc) [reply pkg/artifact/artifacts.go:40]: good idea

Comment (littlejiancc) [reply pkg/artifact/artifacts.go]: @LinuxSuRen done

**Review by LinuxSuRen — APPROVED** [review]

**Pull request merged**